DH-13

Dyrk84 opened the pull request:
Boss in the game!
It is possible to call monsters (goblin and goblin king) via the String type parameter.
The monsterCaller method now can call the goblin king too.
If the goblin king is comes, the champion takes damage.
Running away action don't working versus goblin king.
If the goblin king dies, the champion wins.
The champion gets a healing potion at the start.
Added 8 tests.

Dyrk84 committed:
DH-13
Boss in the game!
It is possible to call monsters (goblin and goblin king) via the String type parameter.
The monsterCaller method now can call the goblin king too.
If the goblin king is comes, the champion takes damage.
Running away action don't working versus goblin king.
If the goblin king dies, the champion wins.
The champion gets a healing potion at the start.
Added 8 tests.

gergelyszanto committed:
DH-13 add monster factory; add CharacterTypes enum

Dyrk84 committed:
DH-13
2 unusable tests

gergelyszanto committed:
DH-13 add MonstersInterface setter

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -22,11 +27,16 @@
+    @Getter
+    private boolean runningAwayFromGoblinKingTest;
+
+    @Getter
+    private boolean goblinKingIsDeadTest;
```
You don't need this. You can get this from monster.isDefeat().

Dyrk84 replied:
i removed

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -36,72 +46,74 @@ public Fight(int numOfMonsters) {  //for test
+        System.out.println("The " + monster.getType().charType + " have now " + monster.getHp() + " hit points");
+        if (monster.getType().equals(CharacterTypes.GOBLIN_KING) && monster.getHp() <= 0) {
+            System.out.println("The goblin king is dead! You win!");
+            goblinKingIsDeadTest = true; //toTest
```
You can test goblin king defeat without this.

Dyrk84 replied:
Tests reworked. This boolen removed

gergelyszanto commented on src/main/java/hu/dungeonhunter/model/CharacterTypes.java:
```diff
@@ -0,0 +1,15 @@
+package hu.dungeonhunter.model;
+
+import lombok.Getter;
```
You can remove this import, it is not used.

Dyrk84 replied:
Thank you! Removed!

Dyrk84 committed:
DH-13
corrected tests

Dyrk84 committed:
DH-13
corrected tests

Dyrk84 committed:
DH-13
final DH-13 version (i hope that :D )

Dyrk84 committed:
DH-13
final DH-13 version (i hope that :D )

Dyrk84 committed:
DH-13
GoblinkKingDeadWinTest renamed to GoblinKingIsDefeatedTest. This test is reworked.
